IndicoToolkit/Extractions.cs

```diff
@@ -141,17 +204,124 @@ public static bool isManuallyAddedPrediction(Prediction prediction)
         /// <summary>
         /// Method <c>selectMaxConfidence</c> gets the highest confidence prediction for a given field.
         /// </summary>
-        internal Prediction selectMaxConfidence(string label) {
-            /* TODO */
-            return null;
+        private Prediction selectMaxConfidence(string label) {
+            Prediction maxPred = null;
+            float confidence = 0f;
+            List<Prediction> preds = PredictionLabelMap[label];
+            for (int i = 0; i < preds.Count; i++)
+            {
+                Prediction pred = preds[i];
+                string predLabel = pred.getLabel();
+                float predConfidence = pred.getSpecialValue($"confidence.{predLabel}");
+                if (predConfidence >= confidence)
+                {
+                    maxPred = pred;
+                    confidence = predConfidence;
+                }
+            }
+            return maxPred;
+        }
+
+        /// <summary>
+        /// Get count of occurrences of each label
+        /// </summary>
+        public Dictionary<string, int> labelCountDict()
+        {
+            Dictionary<string, int> countDict = new Dictionary<string, int>();
+            foreach (var label in LabelSet)
+            {
+                int count = PredictionLabelMap[label].Count;
+                countDict[label] = count;
+            }
+            return countDict;
+        }
+
+        /// <summary>
+        /// Check whether there are multiple unique text values for field
+        /// </summary>
+        public bool existMultipleValsForLabel(string label)
+        {
+            int count = PredictionLabelMap[label].Distinct().ToList().Count;
```

Comment:
wouldn't this count multiple distinct if the same text occurs twice, but with different confidences?